dsa-solutions/lc-solutions/0200-0299/0246-Strobogrammatic-Number.md

```diff
@@ -0,0 +1,315 @@
+2. Two pointers, `left` and `right`, traverse the string from both ends towards the center.
+3. For each pair of digits, we check if they exist in the map and match the required strobogrammatic counterpart.
+4. If any pair is invalid, return false; if all pairs are valid, return true.
+
+</TabItem>
+
+<TabItem value="python" label="Python"> 
+
+1. The dictionary `strobogrammatic_map` defines the valid strobogrammatic pairs.
+2. Two pointers, `left` and `right`, are used to traverse the number from both ends.
+3. For each pair of digits, we check if they are in the dictionary and match the required strobogrammatic counterpart.
+4. If any pair is invalid, return false; if all pairs are valid, return true.
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+
+1. The `strobogrammaticMap` defines the valid strobogrammatic pairs.
+2. Two pointers, `left` and `right`, traverse the string from both ends towards the center.
+3. For each pair of digits, we check if they exist in the map and match the required strobogrammatic counterpart.
+4. If any pair is invalid, return false; if all pairs are valid, return true.
+
+</TabItem>
+
+<TabItem value="cpp" label="C++">
+
+1. The `strobogrammaticMap` defines the valid strobogrammatic pairs.
+2. Two pointers, `left` and `right`, traverse the string from both ends towards the center.
+3. For each pair of digits, we check if they exist in the map and match the required strobogrammatic counterpart.
+4. If any pair is invalid, return false; if all pairs are valid, return true.
+
+</TabItem>
+</Tabs>
+
+
+#### Complexity:
+<Tabs>
+
+<TabItem value="javascript" label="Javascript">
+
+1. **Time complexity**: \(O(n)\), where \(n\) is the length of the input string, as we only iterate through the string once.
+2. **Space complexity**: \(O(1)\), as we use a fixed amount of extra space for the map and pointers.
+3. This makes the approach efficient and suitable for checking strobogrammatic properties in linear time.
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+1. **Time complexity**: \(O(n)\), where \(n\) is the length of the input string, as we only iterate through the string once.
+2. **Space complexity**: \(O(1)\), as we use a fixed amount of extra space for the map and pointers.
+3. This makes the approach efficient and suitable for checking strobogrammatic properties in linear time.
+
+</TabItem>
+
+<TabItem value="python" label="Python"> 
+
+1. Time complexity: \(O(n)\), where \(n\) is the length of the input string, as we only iterate through the string once.
+2. Space complexity: \(O(1)\), as we use a fixed amount of extra space for the dictionary and pointers.
+3. This makes the approach efficient and suitable for checking strobogrammatic properties in linear time.
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+
+1. **Time complexity**: \(O(n)\), where \(n\) is the length of the input string, as we only iterate through the string once.
```

Comment:
replace `\(O(n)\)` with `$O(n)$`  you need to use the right format So, now update all based on our structure and formatting